substrate/primitives/runtime/src/generic/checked_extrinsic.rs

```diff
@@ -18,79 +18,116 @@
+	pub format: ExtrinsicFormat<AccountId, Extension>,
 
 	/// The function that should be called.
 	pub function: Call,
 }
 
-impl<AccountId, Call, Extra, RuntimeOrigin> traits::Applyable
-	for CheckedExtrinsic<AccountId, Call, Extra>
+impl<AccountId, Call, Extension, RuntimeOrigin> traits::Applyable
+	for CheckedExtrinsic<AccountId, Call, Extension>
 where
 	AccountId: Member + MaybeDisplay,
-	Call: Member + Dispatchable<RuntimeOrigin = RuntimeOrigin>,
-	Extra: SignedExtension<AccountId = AccountId, Call = Call>,
+	Call: Member + Dispatchable<RuntimeOrigin = RuntimeOrigin> + Encode,
+	Extension: TransactionExtension<Call, ()>,
 	RuntimeOrigin: From<Option<AccountId>>,
 {
 	type Call = Call;
 
-	fn validate<U: ValidateUnsigned<Call = Self::Call>>(
+	fn validate<I: ValidateUnsigned<Call = Self::Call>>(
 		&self,
-		// TODO [#5006;ToDr] should source be passed to `SignedExtension`s?
-		// Perhaps a change for 2.0 to avoid breaking too much APIs?
 		source: TransactionSource,
 		info: &DispatchInfoOf<Self::Call>,
 		len: usize,
 	) -> TransactionValidity {
-		if let Some((ref id, ref extra)) = self.signed {
-			Extra::validate(extra, id, &self.function, info, len)
-		} else {
-			let valid = Extra::validate_unsigned(&self.function, info, len)?;
-			let unsigned_validation = U::validate_unsigned(source, &self.function)?;
-			Ok(valid.combine_with(unsigned_validation))
+		match self.format {
+			ExtrinsicFormat::Bare => {
+				let inherent_validation = I::validate_unsigned(source, &self.function)?;
+				#[allow(deprecated)]
+				let legacy_validation = Extension::validate_bare_compat(&self.function, info, len)?;
+				Ok(legacy_validation.combine_with(inherent_validation))
+			},
+			ExtrinsicFormat::Signed(ref signer, ref extension) => {
+				let origin = Some(signer.clone()).into();
+				extension.validate_only(origin, &self.function, info, len).map(|x| x.0)
+			},
+			ExtrinsicFormat::General(ref extension) =>
+				extension.validate_only(None.into(), &self.function, info, len).map(|x| x.0),
 		}
 	}
 
-	fn apply<U: ValidateUnsigned<Call = Self::Call>>(
+	fn apply<I: ValidateUnsigned<Call = Self::Call>>(
 		self,
 		info: &DispatchInfoOf<Self::Call>,
 		len: usize,
 	) -> crate::ApplyExtrinsicResultWithInfo<PostDispatchInfoOf<Self::Call>> {
-		let (maybe_who, maybe_pre) = if let Some((id, extra)) = self.signed {
-			let pre = Extra::pre_dispatch(extra, &id, &self.function, info, len)?;
-			(Some(id), Some(pre))
-		} else {
-			Extra::pre_dispatch_unsigned(&self.function, info, len)?;
-			U::pre_dispatch(&self.function)?;
-			(None, None)
-		};
-		let res = self.function.dispatch(RuntimeOrigin::from(maybe_who));
-		let post_info = match res {
-			Ok(info) => info,
-			Err(err) => err.post_info,
-		};
-		Extra::post_dispatch(
-			maybe_pre,
-			info,
-			&post_info,
-			len,
-			&res.map(|_| ()).map_err(|e| e.error),
-		)?;
-		Ok(res)
+		match self.format {
+			ExtrinsicFormat::Bare => {
+				I::pre_dispatch(&self.function)?;
```

Comment:
No because this call is to [`ValidateUnsigned::pre_dispatch`](https://github.com/paritytech/polkadot-sdk/blob/f5b8d1f3c04485166e052267ea16be2b603556eb/substrate/primitives/runtime/src/traits/mod.rs#L1730). According to the docs:
> the `pre_dispatch` function is called right before dispatching the call wrapped by an unsigned extrinsic

And the lines following this one are responsible for the actual dispatch of the function.